tests/test_floyd_warshall.py

```diff
@@ -111,12 +111,15 @@ def test_weighted_numpy_negative_cycle(self):
             (3, 0, -1),
         ])
         dist = retworkx.graph_floyd_warshall_numpy(graph, lambda x: x)
-        expected = numpy.array(
-            [[-6, -7, -8, -13],
-             [-7, -8, -9, -14],
-             [-8, -9, -10, -15],
-             [-13, -14, -15, -20]], dtype=numpy.float64)
-        self.assertTrue(numpy.array_equal(dist, expected))
```

Comment:
Brief explanation why this test is implementation dependent and shouldn't be there:

1. If there's a negative cycle, the shortest path is not well defined. You can loop forever in the cycle and go as negative as you want making the path as short as you want.

So, take for example if we changed the nested inner loop we have for each `k`:
```
for i in 0 .. N {
  for j in 0 .. N {
  /* minimize m */
  }
}
```

For one where we swap `i` and `j`:
```
for j in 0 .. N {
  for i in 0 .. N {
  /* minimize m */
  }
}
```
Both ways of writing the loops visit all pairs `mat[i][j]`. For graphs without negative cycles, swapping `i` and `j` doesn't change correctness.

But for graphs with negative cycles, we would visit those negative cycles in a different order if we swap `i` and `j`. So the matrix results that would be slightly different. Even though the matrix results are meaningless because there's no shortest path!

Hence the reason to remove the matrix comparison